packages/clay-core/src/typography/Text.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * SPDX-FileCopyrightText: © 2022 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React, {ElementType} from 'react';
+
+export type TextSize = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+
+export type DisplayType = 'p' | 'span';
+
+export type WeightFont =
+	| 'lighter'
+	| 'light'
+	| 'normal'
+	| 'semi-bold'
+	| 'bold'
+	| 'bolder';
+
+export type ColorType =
+	| null
+	| 'primary'
+	| 'secondary'
+	| 'muted'
+	| 'success'
+	| 'warning'
+	| 'danger'
+	| 'info';
+
+type Props = {
+	/**
+	 * Determine the way in which a text is displayed.
+	 */
+
+	as?: DisplayType;
```

Comment:
Mmmm, so if I understood correctly the idea is to import the inline type from `Heading` prop component `as`? If it is, the problem is that in `Heading` component we only use the `h1` tag but in `Text` component we use the `p` and `span`. For that reason, that's why I created a specific type because in `Heading` there is only one tag.

Reply:
Oh sorry, that's not what I meant, in this case, I'm saying that instead of creating `type DisplayType` separately from `type Props` in this case `as`, you can define `type` inline instead of creating one separately, so in that case, you would just define.

```
as?: 'p' | 'span';
```

Even if the declaration is inline it is still possible to access it if someone wants to reuse it. The problem with exporting these types is that they are generic, ie `DisplayType` is correct when it is used only in that file and not exported because the understood that is referring to this file but when it is exported, there may be a name conflict when other files do the same, then we would have to adopt a certain default in the future, eg `TextDisplayType` to avoid name collision.